### PR TITLE
Using UID instead of title to identify runner, suite, spec and steps

### PR DESCRIPTION
**Pull request by wvankuipers**
**Please note:** this PR is required for fixing issue [#11 reported in the wdio-spec-reporter](https://github.com/webdriverio/wdio-spec-reporter/issues/11) repo, please refer to that issue for more details.

## Proposed changes

I implemented an UID to identify suite/spec/steps to prevent duplicate named items forming an issue when reported back to the wdio-spec-reporter.

## Types of changes

What types of changes does your code introduce to WebdriverIO?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) *added fallback for reporters that don't report a UID*

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works **There seems to be no tests for the stats reporter util?**
- [ ] I have added necessary documentation (if appropriate) **There seems to be no documentation about the stats reporter util?**

## Further comments

For further information about this PR please refer to the issue mentioned above.

### CODE REVIEW

File: lib/utils/ReporterStats.js [hunk 1]
```diff
@@ -207,13 +212,13 @@ class ReporterStats extends RunnableStats {
 
         // Errors encountered inside hooks (e.g. beforeEach) can be identified by looking
         // at the currentTest param (currently only applicable to the Mocha adapter).
-        let title = runner.title
-        if (!suiteStats.hooks[title]) {
-            title = runner.title
+        let uid = runner.currentTest || runner.uid
+        if (!suiteStats.tests[uid]) {
```
Comment (christian-bromann) [inline lib/utils/ReporterStats.js]: why `suiteStats.tests` if it is `getHookStats`?

Comment (wvankuipers) [inline lib/utils/ReporterStats.js]: Yeah good catch, it was a incorrect call, corrected it

**Comment by wvankuipers** [comment]
We could use a getIdentifier method:

    getIdentifier: function (runner) {
        return runner.uid || runner.title;
    }

That should cover both the old and the new scenario.

**Comment by christian-bromann** [comment]
@wvankuipers let's do this

**Comment by christian-bromann** [comment]
@wvankuipers I think this will also fix #1742 right?

**Comment by wvankuipers** [comment]
> I think this will also fix #1742 right?

Yes!

**Comment by christian-bromann** [comment]
@wvankuipers can you check CI errors? I think they are legit:

```
Uncaught TypeError: _this.getIdentifier is not a function
```

File: lib/utils/ReporterStats.js [hunk 2]
```diff
@@ -327,16 +327,16 @@ class ReporterStats extends RunnableStats {
...
 
     runnerEnd (runner) {
         this.getSpecStats(runner).complete()
     }
+}
 
-    getIdentifier (runner) {
-        return runner.uid || runner.title
-    }
+function getIdentifier (runner) {
```
Comment (christian-bromann) [inline lib/utils/ReporterStats.js]: @wvankuipers Can this method be a static method in `RunnableStats`?

**Pull request merged**